pkg/apis/core/types_shoot.go

```diff
@@ -678,6 +678,9 @@ type OIDCConfig struct {
 	// If set, the OpenID server's certificate will be verified by one of the authorities in the oidc-ca-file, otherwise the host's root CA set will be used.
 	CABundle *string
 	// ClientAuthentication can optionally contain client configuration used for kubeconfig generation.
+	// Deprecated: This field has no implemented use.
+	// It's use was planned for genereting OIDC kubeconfig https://github.com/gardener/gardener/issues/1433
+	// TODO(AleksandarSavchev): Drop this field after v1.102 has been released.
```

Comment:
Even if it didn't have a use before, I think we should be more conservative here and remove it only after at least 3 months (= 6 releases). WDYT?

Reply:
Actually, according to https://github.com/gardener/gardener/blob/master/docs/development/process.md#deprecations-and-backwards-compatibility, let's rather forbid to set this field for `1.31+` shoots. This way, we are on the safe side to not break any users if they set this field.